testserver/testserver.go

```diff
@@ -403,8 +415,8 @@ func NewTestServer(opts ...TestServerOpt) (TestServer, error) {
 
 	if serverArgs.cockroachBinary != "" {
 		// CockroachBinaryPathOpt() overrides the flag or env variable.
-	} else if len(*customBinaryFlag) > 0 {
-		serverArgs.cockroachBinary = *customBinaryFlag
+	} else if len(getCustomBinaryFlag()) > 0 {
```

Comment:
Won't we need to have a setter too so we can set the flag with lock-acquired?

At the moment the flag is set this way:
```
f := flag.Lookup("cockroach-binary")
	if f == nil {
		return errors.New("flag cockroach-binary not found")
	}
	if err := f.Value.Set(cockroach); err != nil {
		return errors.Wrap(err, "could not set value of cockroach-binary flag")
	}
```
but then won't lock-checking identify writes that do not acquire lock in violation?

Reply:
I assumed that the write will happen at startup, before any test tarts. But yes, for completeness, we must lock it here as well. I will update the PR.

Reply:
I don't think race-checker would see it as a life-cycle thing.